app/controllers/course/forum/posts_controller.rb

```diff
@@ -0,0 +1,56 @@
+class Course::Forum::PostsController < Course::Forum::Controller
+  include Course::Discussion::PostsConcern
+
+  def create
+    if super
+      redirect_to course_forum_topic_path(current_course, @forum, @topic),
+                  success: t('course.discussion.posts.create.success')
+    else
+      redirect_to course_forum_topic_path(current_course, @forum, @topic),
+                  danger: t('course.discussion.posts.create.failure',
+                            error: @post.errors.full_messages.to_sentence)
+    end
+  end
+
+  def update
+    if super
+      redirect_to course_forum_topic_path(current_course, @forum, @topic),
+                  success: t('course.discussion.posts.update.success')
+    else
+      render 'edit'
+    end
+  end
+
+  def destroy
+    if super
+      redirect_to course_forum_topic_path(current_course, @forum, @topic),
+                  success: t('course.discussion.posts.destroy.success')
+    else
+      redirect_to course_forum_topic_path(current_course, @forum, @topic),
+                  danger: t('course.discussion.posts.destroy.failure',
+                            error: @post.errors.full_messages.to_sentence)
+    end
+  end
+
+  protected
+
+  def auto_subscribe_topic
+    @topic.subscriptions.build(user: current_user).save unless @topic.subscribed_by?(current_user)
```

Comment:
You should provide a model method instead of building a subscription like this in the controller.